impeller/renderer/backend/vulkan/command_encoder_vk.cc

```diff
@@ -124,13 +124,23 @@ bool CommandEncoderVK::IsValid() const {
   return is_valid_;
 }
 
-bool CommandEncoderVK::Submit() {
+bool CommandEncoderVK::Submit(SubmitCallback callback) {
+  // Make sure to call callback with `false` if anything returns early.
+  bool fail_callback = !!callback;
```

Comment:
Prefer a ScopedCleanupClosure that returns false by default. On the one success case, release the closure and invoke it with true once.

Reply:
That's being done below.

Reply:
`fail_callback` is used in the ScopedCleanupClosure.

Reply:
Oh, I think I see what you mean.

Reply:
I thought there was some way to make this a little clearer with a second fml::ScopedCleanupClosure but I'm missing it. Maybe we can sync up offline.

Reply:
Argh, nevermind again, I misread this and now I understand what you're saying :)